### ISSUE TITLE
Check that `StatesSaved` matches `StatesCurrent` before running `EnsureCmd::exec` / `CleanCmd::exec`

### ISSUE THREAD

**Issue by azriel91** [azriel91/peace#86]
Similar to `git push`, an apply command should not alter state if the user made the decision based on stale saved state.

Perhaps do this after #67, so it only has to do be done once.

**Comment by azriel91** [comment]
Duplicate of #59.
